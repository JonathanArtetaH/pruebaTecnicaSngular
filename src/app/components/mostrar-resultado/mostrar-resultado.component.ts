import { Component, Input} from '@angular/core';

@Component({
  selector: 'app-mostrar-resultado',
  template: `
  <h3>Resultado es: {{resultado}}</h3> 
  <p *ngIf="resultado!=0">serie ({{numeroInput}}): ( 2primo({{numeroInput}}-2) ) * ( 3triangular({{numeroInput}}-2) ) * ( 7fibonacci({{numeroInput}}-1) )</p>
  `,
  styles: ['']
})

export class MostrarResultadoComponent {
  @Input() resultado: number = 0; 
  @Input() numeroInput: number = 0;  

}
