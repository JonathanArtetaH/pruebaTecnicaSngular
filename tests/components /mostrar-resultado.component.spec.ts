import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';    
import { MostrarResultadoComponent } from '../../src/app/components/mostrar-resultado/mostrar-resultado.component';

describe('MostrarResultadoComponent', () => {
  
  let component: MostrarResultadoComponent;
  let fixture: ComponentFixture<MostrarResultadoComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        MostrarResultadoComponent
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(MostrarResultadoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('Creacion de componente correctamente', () => {
    expect(component).toBeTruthy();
  });

  test('Debe coincidir con el snapshot', () => {
    const fixture = TestBed.createComponent(MostrarResultadoComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect( compiled ).toMatchSnapshot();
  })

   
});
