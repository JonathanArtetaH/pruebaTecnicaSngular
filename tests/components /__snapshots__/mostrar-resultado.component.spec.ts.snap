// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostrarResultadoComponent Debe coincidir con el snapshot 1`] = `
<div
  id="root2"
>
  <h3>
    Resultado es: 0
  </h3>
  
</div>
`;
